app/src/main/java/org/wikipedia/util/StringUtil.java

```diff
@@ -261,4 +264,16 @@ public static String listToJSONString(@Nullable List<Integer> list) {
 
     private StringUtil() {
     }
+
+    public static ArrayList<String> extractUrlsFromhtmlString(String html) {
+        ArrayList<String> result = new ArrayList<>();
+        String regex = "(https)://[-a-zA-Z0-9+&@#/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|]";
+        Pattern pattern = Pattern.compile(regex);
+        Matcher matcher = pattern.matcher(html);
```

Comment:
Is it necessary to create a new `Pattern`? It would be better if does the following: 
`Matcher matcher = Pattern.compile(regex).matcher(html);`

Reply:
I feel things like this make code easily readable, but I have changed it anyway.